File: samples/java-play2/app/security/AuthorisationFilter.java
package security;

import com.wordnik.swagger.model.*;
import com.wordnik.swagger.core.filter.SwaggerSpecFilter;

import org.slf4j.*;

import java.util.Map;
import java.util.List;

public class AuthorisationFilter implements SwaggerSpecFilter {
    static Logger logger = LoggerFactory.getLogger("swagger");

    public boolean isOperationAllowed(Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
        boolean isAuthorized = checkKey(params, headers);
        if (isAuthorized) {
            return true;
        } else {
            if (operation.method().equals("GET")) {
                if (api.path().startsWith("/store")) {
                    return false;
                } else {
                    return true;
                }
            } else {
                return false;
            }
        }
    }

    public boolean isParamAllowed(Parameter parameter, Operation operation, ApiDescription api, Map<String, List<String>> params, Map<String, String> cookies, Map<String, List<String>> headers) {
        boolean isAuthorized = checkKey(params, headers);
        if ((parameter.paramAccess().isDefined() && parameter.paramAccess().get().equals("internal")) && !isAuthorized)
            return false;
        else
            return true;
    }

    public boolean checkKey(Map<String, List<String>> params, Map<String, List<String>> headers) {
        String keyValue = null;
        if (params.containsKey("api_key"))
            keyValue = params.get("api_key").get(0);
        else {
            if (headers.containsKey("api_key"))
                keyValue = headers.get("api_key").get(0);
        }
        if ("special-key".equals(keyValue))
            return true;
        else
            return false;
    }
}

File: samples/scala-play2/app/security/AuthorisationFilter.scala
package security

import com.wordnik.swagger.model._
import com.wordnik.swagger.core.filter.SwaggerSpecFilter
import play.api.Logger

class AuthorisationFilter extends SwaggerSpecFilter {

  def isOperationAllowed(operation: Operation, api: ApiDescription, params: java.util.Map[String, java.util.List[String]], cookies: java.util.Map[String, String], headers: java.util.Map[String, java.util.List[String]]): Boolean = {
    checkKey(params, headers) match {
      case true => true
      case false => {
        Logger("swagger").debug("authenticated: false  - method: %s - path: %s".format(operation.method, api.path))
        if(operation.method == "GET" && api.path.indexOf("/admin") != -1) true
        else if(operation.method == "GET" && api.path.indexOf("/pet") != -1) true
        else if(operation.method == "GET" && api.path.indexOf("/user") != -1) true
        else false
      }
    }
  }

  def isParamAllowed(parameter: Parameter, operation: Operation, api: ApiDescription, params: java.util.Map[String, java.util.List[String]], cookies: java.util.Map[String, String], headers: java.util.Map[String, java.util.List[String]]): Boolean = {
    val isAuthorized = checkKey(params, headers)  
    if(parameter.paramAccess == Some("internal") && !isAuthorized) false
    else true
  }

  def checkKey(params: java.util.Map[String, java.util.List[String]], headers: java.util.Map[String, java.util.List[String]]): Boolean = {
    val apiKey = params.containsKey("api_key") match {
      case true => Some(params.get("api_key").get(0))
      case _ => {
        headers.containsKey("api_key") match {
          case true => Some(headers.get("api_key").get(0))
          case _ => None
        }
      }
    }

    apiKey match {
      case Some(key) if(key == "special-key") => true
      case _ => false
    }
  }
}